images/router/haproxy/conf/haproxy-config.template

```diff
@@ -606,20 +546,8 @@ backend be_tcp:{{$cfgIdx}}
           "<cert>: <domain-set>" is important as this allows us to use
          wildcards and/or use a deny set with !<domain> in the future.
 */}}
-{{ define "/var/lib/haproxy/conf/.tmp/cert_config.map" -}}
-{{     $workingDir := .WorkingDir -}}
-{{     range $idx, $cfg := .State -}}
-{{       if and (ne $cfg.Host "") (matchValues (print $cfg.TLSTermination) "edge" "reencrypt") -}}
-{{         $cert := index $cfg.Certificates $cfg.Host -}}
-{{         if ne $cert.Contents "" -}}
-{{$workingDir}}/certs/{{$idx}}.pem {{genCertificateHostName $cfg.Host $cfg.IsWildcard}}
-{{         end -}}
-{{      end -}}
-{{     end -}}
-{{ end }}{{/* end temporary cert_config map template */}}
-
 {{ define "/var/lib/haproxy/conf/cert_config.map" -}}
-{{     range $idx, $data := sortedMapData "/var/lib/haproxy/conf/.tmp/cert_config.map" false -}}
-{{ $data }}
+{{     range $idx, $line := generateHAProxyMap "cert_config.map" . -}}
```

Comment:
I like this a lot more

Reply:
Although....

Can you get at the template name from within a template?  If we did that we could avoid having the filename argument.  I.e. when you create the template create a closure over the template name.

Reply:
Basically, in pkg/router/template/plugin.go line 122 (inside NewTemplatePlugin) when we iterate over the loop of templates, create a closure on the specific helpers that are filename specific and create a closure with:

```
for _, template := range masterTemplates.Templates() {
  ...
  // copy helperFunctions into a new map
  name := template.Name()
  fns["generateHAProxyMap"] = func(State) {
    // use name here instead of taking it as an argument
  }
  // do the same for others
  templates.Funcs(
  templates[template.Name()] = template
}
```

Reply:
ok will do.